packages/http-specs/specs/response/status-code-range/main.tsp

```diff
@@ -0,0 +1,44 @@
+import "@typespec/http";
+import "@typespec/spector";
+
+using Http;
+using Spector;
+
+/**
+ * Test for range of status code.
+ */
+@scenarioService("/response/status-code-range")
+namespace Response.StatusCodeRange;
+
+@scenario
+@scenarioDoc("""
+  Test case for range of status code in error response.
+
+  Verify that the result of the API is an error/exception in client, and the error response can be de-serialized to ErrorInRange model (instead of DefaultError model).
+
+  Expected response body:
+  status code 494
+  ```json
+  { 
+    "code": "request-header-too-large",
+    "message": "Request header too large"
+  }
+  ```
+  """)
+@route("/error-response")
+@get
+op errorResponse(): 204 | ErrorInRange | DefaultError;
```

Comment:
I can remove `DefaultError`, as this one would not affect the code in range (494-499).

Reply:
@weidongxu-microsoft Is it possible to add the parameter to trigger different response error? For example:
`op errorResponse(@query statusCode: int)` then mock server could return the related error model with the statusCode of request body.

Reply:
Given the "MockApiDefinition" pattern, it be simpler to have concrete request/response for one operation (otherwise we will need to add a `handler`, and it may have other complications).

Let's see whether we have this ask from another language. If this is a broader ask, I will update the op.

Reply:
I don't think JS has this requirement. Concrete error is fine with me.